### DISCUSSION TITLE
How to use type-graphql with vite in node?

### DISCUSSION THREAD

**Discussion by Iran-110** [MichalLytek/type-graphql#1769] [Q&A]
How to configure using vite-plugin-node to properly enable hot module replacement?

**Comment by Iran-110** [comment]
There seem to be issues when applying HMR such as redefining `decorators` and inconsistencies in `metadata`.

**Comment by MichalLytek** [comment]
In next.js HMR I do the trick of using custom webpack plugin that clears metadata storage. This prevents memory leaks that would slow down the app.

Haven't used Vite so can't help you with that.

**Comment by cjnoname** [comment]
Use swc-node or any SWC-supported libraries.
